File: CS/TestConditionalAppearanceRules.Module/Person.cs
using DevExpress.ExpressApp.Model;
using System;
using System.ComponentModel;

using DevExpress.Xpo;
using DevExpress.Data.Filtering;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.Validation;
using DevExpress.ExpressApp.ConditionalAppearance;
using DevExpress.Persistent.BaseImpl;

namespace TestConditionalAppearanceRules.Module {
    [DefaultClassOptions, ImageName("BO_Person"), ModelDefault("Caption", "Person")]
    public class MyPerson : BaseObject {
        public MyPerson(Session session) : base(session) { }
        private string name;
        [Appearance("NameFontIsRed", Criteria="IsMarried", FontColor="Red")]
        [Appearance("NameFontIsGreen", Criteria="!IsMarried", FontColor = "Green")]
        public string Name {
            get { return name; }
            set { SetPropertyValue("Name", ref name, value); }
        }
        private bool isMarried;
        [ImmediatePostData]
        public bool IsMarried {
            get { return isMarried; }
            set { 
                SetPropertyValue("IsMarried", ref isMarried, value);
                if (!String.IsNullOrEmpty(SpouseName) && !value) SpouseName = String.Empty;
            }
        }
        private string spouseName;
        [Appearance("DisableSpouseName", Criteria="!IsMarried", Enabled=false)]
        public string SpouseName {
            get { return spouseName; }
            set { SetPropertyValue("SpouseName", ref spouseName, value); }
        }
    }
}

File: VB/TestConditionalAppearanceRules.Module/Person.vb
Imports DevExpress.ExpressApp.Model
Imports System
Imports System.ComponentModel

Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering

Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.Validation
Imports DevExpress.ExpressApp.ConditionalAppearance
Imports DevExpress.Persistent.BaseImpl

Namespace TestConditionalAppearanceRules.Module
    <DefaultClassOptions, ImageName("BO_Person"), ModelDefault("Caption", "Person")> _
    Public Class MyPerson
        Inherits BaseObject

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        Private name_Renamed As String
        <Appearance("NameFontIsRed", Criteria:="IsMarried", FontColor:="Red"), Appearance("NameFontIsGreen", Criteria:="!IsMarried", FontColor := "Green")> _
        Public Property Name() As String
            Get
                Return name_Renamed
            End Get
            Set(ByVal value As String)
                SetPropertyValue("Name", name_Renamed, value)
            End Set
        End Property

        Private isMarried_Renamed As Boolean
        <ImmediatePostData> _
        Public Property IsMarried() As Boolean
            Get
                Return isMarried_Renamed
            End Get
            Set(ByVal value As Boolean)
                SetPropertyValue("IsMarried", isMarried_Renamed, value)
                If (Not String.IsNullOrEmpty(SpouseName)) AndAlso (Not value) Then
                    SpouseName = String.Empty
                End If
            End Set
        End Property

        Private spouseName_Renamed As String
        <Appearance("DisableSpouseName", Criteria:="!IsMarried", Enabled:=False)> _
        Public Property SpouseName() As String
            Get
                Return spouseName_Renamed
            End Get
            Set(ByVal value As String)
                SetPropertyValue("SpouseName", spouseName_Renamed, value)
            End Set
        End Property
    End Class
End Namespace
